Mathlib/Data/Set/Quot.lean

```diff
@@ -0,0 +1,77 @@
+/-
+Copyright (c) 2025 Pim Otte. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Pim Otte
+-/
+
+import Mathlib.Data.Set.Card
+
+/-!
+# Quotients as Sets
+
+## Main definitions
+
+* `Quot.toSet` interprets an equivalence class as a set.
+* `Set.Represents` says that a set represents a set of quotients if it contains exactly one element
+  from each quotient.
+-/
+
+variable {α : Type*} {r : α → α → Prop}
+
+/-- A quotient element can be interpreted as the set of elements in the same equivalence class. -/
+instance : SetLike (Quot r) α where
+  coe (c : Quot r) := {x | Quot.mk r x = c}
+  coe_injective' x y h := by
+    dsimp at h
+    simpa using show x.out ∈ {v | Quot.mk r v = y} by
+      rw [← h]
+      exact x.out_eq
+
+@[simp] lemma Quot.mem_coe {x : α} {c : Quot r} : x ∈ c ↔ Quot.mk r x = c := Iff.rfl
+
+@[ext] theorem Quot.ext (c d : Quot r) (h : ∀ x, x ∈ c ↔ x ∈ d) : c = d := SetLike.ext h
+
+namespace Set
+
+/-- A set represents a set of quotients if it contains exactly one element from each quotient. -/
+def Represents (s : Set α) (C : Set (Quot r)) := Set.BijOn (Quot.mk r) s C
+
+lemma out_image_represents (C : Set (Quot r)) : (Quot.out '' C).Represents C :=
```

Comment:
Is there a general result about `BijOn` and functions which satisfy `mk (out x) = x` (`RightInverse`) here?